akka-cluster-sharding-typed/src/multi-jvm/scala/akka/cluster/sharding/ClusterShardingRememberEntitiesPerfSpec.scala

```diff
@@ -0,0 +1,306 @@
+  case class Out(latency: Long)
+
+  class LatencyEntity extends Actor with ActorLogging {
+
+    override def receive: Receive = {
+      case In(_, created) =>
+        sender() ! Out(System.currentTimeMillis() - created)
+      case _: Stop =>
+//        log.debug("Stop received {}", self.path.name)
+        context.parent ! Passivate("stop")
+      case "stop" =>
+//        log.debug("Final Stop received {}", self.path.name)
+        context.stop(self)
+      case msg => throw new RuntimeException("unexpected msg " + msg)
+    }
+  }
+
+  object LatencyEntity {
+
+    val extractEntityId: ShardRegion.ExtractEntityId = {
+      case in: In         => (in.id.toString, in)
+      case msg @ Stop(id) => (id.toString, msg)
+    }
+
+    val extractShardId: ShardRegion.ExtractShardId = _ => "0"
+  }
+
+  case class Stop(id: Int)
+
+}
+
+object ClusterShardingRememberEntitiesPerfSpecConfig
+    extends MultiNodeClusterShardingConfig(
+      rememberEntities = true,
+      additionalConfig = s"""
+    akka.loglevel = DEBUG 
+    akka.testconductor.barrier-timeout = 3 minutes
+    akka.remote.artery.advanced.outbound-message-queue-size = 10000
+    akka.remote.artery.advanced.maximum-frame-size = 512 KiB
+    # comment next line to enable durable lmdb storage
+    akka.cluster.sharding.distributed-data.durable.keys = []
+    akka.cluster.sharding {
+      remember-entities = on
+    }
+    """) {
+
+  val first = role("first")
+  val second = role("second")
+  val third = role("third")
+
+  nodeConfig(third)(ConfigFactory.parseString(s"""
+    akka.cluster.sharding.distributed-data.durable.lmdb {
+      # use same directory when starting new node on third (not used at same time)
+      dir = "$targetDir/sharding-third"
+    }
+    """))
+}
+
+class ClusterShardingRememberEntitiesPerfSpecMultiJvmNode1 extends ClusterShardingRememberEntitiesPerfSpec
+class ClusterShardingRememberEntitiesPerfSpecMultiJvmNode2 extends ClusterShardingRememberEntitiesPerfSpec
+class ClusterShardingRememberEntitiesPerfSpecMultiJvmNode3 extends ClusterShardingRememberEntitiesPerfSpec
+
+abstract class ClusterShardingRememberEntitiesPerfSpec
+    extends MultiNodeClusterShardingSpec(ClusterShardingRememberEntitiesPerfSpecConfig)
+    with ImplicitSender {
+  import ClusterShardingRememberEntitiesPerfSpec._
+  import ClusterShardingRememberEntitiesPerfSpecConfig._
+
+  val nrRegions = 6
+
+  def startSharding(): Unit = {
+    (1 to nrRegions).foreach { n =>
+      startSharding(
+        system,
+        typeName = s"EntityLatency$n",
+        entityProps = Props(new LatencyEntity()),
+        extractEntityId = LatencyEntity.extractEntityId,
+        extractShardId = LatencyEntity.extractShardId)
+    }
+
+  }
+
+  var latencyRegions = Vector.empty[ActorRef]
+
+  // use 5 for "real" testing
+  private val nrIterations = 5
+  // use 5 for "real" testing
+  private val numberOfMessagesFactor = 5
+
+  val latencyCount = new AtomicInteger(0)
```

Comment:
When is this accessed from different threads, could be just a var?

Reply:
Yeah i wasn't sure where to put this